### ISSUE TITLE
ActiveModel::UnknownAttributeError (unknown attribute 'whole_word' for CustomFilter.)

### ISSUE THREAD

**Issue by kescherCode** [glitch-soc/mastodon#1893]
### Steps to reproduce the problem

1. Use the instance as normal
2. Create some replies and statuses (perhaps with active filters?)
3. Watch the logs whenever you get an unexpected, but rare "Oops!" dialog in the web frontend while creating a new status

### Expected behaviour

Nothing

### Actual behaviour

The error described below

### Detailed description

Occasionally, I see the following in the logs:

```
ActiveModel::UnknownAttributeError (unknown attribute 'whole_word' for CustomFilter.):

app/controllers/api/v1/filters_controller.rb:16:in `block in create'
app/controllers/api/v1/filters_controller.rb:15:in `create'
app/controllers/concerns/localized.rb:11:in `set_locale'
lib/mastodon/rack_middleware.rb:9:in `call'
```

This has previously been reported to me as random "Oops!" notification in the web frontend by a few users. Recently, this hit me as well and it seems this is the cause.

The error likely due to `:whole_word` in `resource_params`: https://github.com/glitch-soc/mastodon/blob/2f8fb49d1314db931385089bc9004a48700161ad/app/controllers/api/v1/filters_controller.rb#L55

Note the difference in this line compared to v2:

https://github.com/glitch-soc/mastodon/blob/2f8fb49d1314db931385089bc9004a48700161ad/app/controllers/api/v2/filters_controller.rb#L46

So, this seems to occur because :whole_word is only ever an attribute on `CustomFilterKeyword`s, not on `CustomFilter`s.

### Specifications

2f8fb49d1314db931385089bc9004a48700161ad

**Comment by ClearlyClaire** [comment]
> * Use the instance as normal
> 
> * Create some replies and statuses (perhaps with active filters?)
> 
> * Watch the logs whenever you get an unexpected, but rare "Oops!" dialog in the web frontend while creating a new status

Hm, the error you're reporting has nothing to do with creating a status. There is an error in the legacy filter controller though.

**Comment by ClearlyClaire** [comment]
Fixed by https://github.com/mastodon/mastodon/pull/19878 merged in #1910
